.ci/scripts/dswx_compare_opera_pge.py

```diff
@@ -0,0 +1,223 @@
+    print(f'{flag_same_geotransforms_str}Comparing geotransform')
+    if not flag_same_geotransforms:
+        print(prefix + f'* input 1 geotransform with content "{geotransform_1}"'
+              f' differs from input 2 geotransform with content'
+              f' "{geotransform_2}".')
+
+    # compare metadata
+    metadata_error_message, flag_same_metadata = \
+        _compare_dswx_hls_metadata(metadata_1, metadata_2, metadata_exclude_list=metadata_exclude_list)
+
+    flag_all_ok, flag_same_metadata_str = _get_prefix_str(flag_same_metadata,
+                                             flag_all_ok)
+    print(f'{flag_same_metadata_str}Comparing metadata')
+
+    if not flag_same_metadata:
+        print(prefix + metadata_error_message)
+
+    return flag_all_ok
+
+
+def _compare_dswx_hls_metadata(metadata_1, metadata_2, metadata_exclude_list=[]):
+    """
+    Compare DSWx-HLS products' metadata
+
+    Parameters
+    ----------
+    metadata_1 : dict
+        Metadata of the first DSWx-HLS product
+    metadata_2: dict
+        Metadata of the second
+    metadata_exclude_list: list
+        Metadata keys to ignore for purposes of determining comparison ok or fail status
+
+    Returns
+    -------
+    metadata_error_message: str
+        A string containing any metadata comparison failure messages
+    flag_same_metadata: bool
+        Flag indicating that metadata comparison succeeded
+    """
+    flag_same_metadata = True
+    metadata_error_message = ""
+    for k2, v2, in metadata_2.items():
+        if k2 not in metadata_1.keys():
+            msg = f'* the metadata key {k2} is present in input 2 but it is not present in input 1'
+            if k2 in metadata_exclude_list:
+                print(f'[INFO] {msg}')
+            else:
+                flag_same_metadata = False
+                metadata_error_message += msg + '\n' + PREFIX
+    for k1, v1, in metadata_1.items():
+        if k1 not in metadata_2.keys():
+            msg = f'* the metadata key {k1} is present in input 1 but it is not present in input 2'
+            if k1 in metadata_exclude_list:
+                # We will just print the difference in the output
+                print(f'[INFO] {msg}')
+            else:
+                flag_same_metadata = False
+                metadata_error_message += msg + '\n' + PREFIX
+        else:
+            # Currently these are string values otherwise the comparison below would need to change for floating point.
+            if metadata_2[k1] != v1:
+                msg = (f'* contents of metadata key {k1} from'
+                       f' input 1 has value "{v1}" whereas the same key in'
+                       f' input 2 metadata has value "{metadata_2[k1]}"')
+                # Don't fail for metadata fields that are not required to be the same
+                if k1 in metadata_exclude_list:
+                    # We will just print the difference in the output
+                    print(f'[INFO] {msg}')
```

Comment:
These will print before the "Comparing metadata" message on line 140, which might be confusing.  I suggest just appending these into the `metadata_error_message` string, or making another return argument to contain the info messages, and print them all at the same time.  Probably good to group these together after the other metadata error messages.

Reply:
I am committing changes to address the most-recent comments.
Here is sample output for a few scenarios with the new changes:
[comparison_all_ok.txt](https://github.com/nasa/opera-sds-pge/files/11065747/comparison_all_ok.txt)
[comparison_band_difference_fail.txt](https://github.com/nasa/opera-sds-pge/files/11065748/comparison_band_difference_fail.txt)
[comparison_fail_num_band_geotransform_metadata.txt](https://github.com/nasa/opera-sds-pge/files/11065749/comparison_fail_num_band_geotransform_metadata.txt)